dhis2/src/main/java/org/motechproject/dhis2/rest/service/impl/DhisWebServiceImpl.java

```diff
@@ -84,6 +95,26 @@ public DhisWebServiceImpl(@Qualifier("dhisSettingsService") SettingsService sett
         this.client = httpClientBuilderFactory.newBuilder().build();
     }
 
+    @PostConstruct
+    public void discoverServerVersion() {
+        Settings settings = settingsService.getSettings();
+
+        if (StringUtils.isNotBlank(settings.getServerURI()) && StringUtils.isNotBlank(settings.getUsername()) && StringUtils.isNotBlank(settings.getPassword())) {
```

Comment:
Anonymous access to the DHIS2 Web API is not allowed. (http://dhis2.github.io/dhis2-docs/2.21/en/developer/html/ch01.html) The check for the username/password is done to avoid exception coming from the HttpClient, where we would provide empty login credentials while crafting basic auth header.